lib/galaxy/dependencies/__init__.py

```diff
@@ -320,6 +320,13 @@ def check_pkce(self):
     def check_rucio_clients(self):
         return "rucio" in self.object_stores
 
+    def check_redis(self):
+        celery_enabled = self.config.get("enable_celery_tasks", False)
+        celery_conf = self.config.get("celery_conf") or {}
+        celery_result_backend = celery_conf.get("result_backend") or ""
+        celery_broker_url = celery_conf.get("broker_url") or ""
+        return celery_enabled and celery_result_backend.startswith("redis") or celery_broker_url.startswith("redis")
```

Comment:
```suggestion
        return celery_enabled and celery_result_backend.startswith("redis://") or celery_broker_url.startswith("redis://")
```

Reply:
Ah, I overlooked this part 😅

Reply:
Apparently it can also be `rediss://` (double "s", to connect to redis over TLS) and also just `socket://` https://docs.celeryq.dev/en/stable/userguide/configuration.html#conf-redis-result-backend
Maybe split on `://` and check that the protocol is in `["redis", "rediss", "redis+socket", "socket"]` ?

Reply:
Done in https://github.com/galaxyproject/galaxy/pull/20502/commits/f3ae4b2fd54e1f132f34ff40218aad44b3dd65d0 :+1: